master/buildbot/worker/base.py

```diff
@@ -781,6 +781,8 @@ def buildFinished(self, sb):
         if not self.building:
             if self.build_wait_timeout == 0:
                 d = self.insubstantiate()
+                d.addCallback(lambda _:
+                              AbstractWorker.disconnect(self))
```

Comment:
Is that something insubstantiate should always do, or should it be a kwarg defaulting to False?

I didn't put it in insubstantiate, because I wasn't sure on the intended usage of the method and didn't want to break something else unintentionally.